chromium_src/ios/web/public/web_state.h

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_CHROMIUM_SRC_IOS_WEB_PUBLIC_WEB_STATE_H_
+#define BRAVE_CHROMIUM_SRC_IOS_WEB_PUBLIC_WEB_STATE_H_
+
+#include <string_view>
+
+#define callbacks_                                                           \
+  callbacks_;                                                                \
+                                                                             \
+ public:                                                                     \
+  template <typename Interface>                                              \
+  void AddUntrustedInterface(                                                \
+      std::string_view host,                                                 \
+      base::RepeatingCallback<void(mojo::PendingReceiver<Interface>)>        \
+          callback) {                                                        \
+    CHECK(!host.empty());                                                    \
+    untrusted_callbacks_.emplace(std::string(host), Interface::Name_);       \
```

Comment:
why are we only checking the host here? I think we should stick with GURL for an extra layer of security here, especially since we don't have any tests for this right now.